### PR TITLE
update dependencies to fix various high/critical security vulnerabilies

### PR DESCRIPTION
**Pull request by pc-bob**
### **User description**

This vulnerability scan results below is from the scan of the [0.24-github_app](https://hub.docker.com/layers/codiumai/pr-agent/0.24-github_app/images/sha256-cc054e25091a1a1956771e146d0556e7f28af043c3cfa8fd65a1709b21cb3d03?context=explore) docker image but I believe would apply to most versions of the image. 

```
Security Violations
┌──────────┬──────────────────────────────────────────────────────────────────────────────┬─────────┬────────────┬──────────┬──────────┬────────┬────────────────┬──────┬──────┬─────────────┐
│ SEVERITY │ DIRECT                                                                       │ DIRECT  │ IMPACTED   │ IMPACTED │ FIXED    │ TYPE   │ CVE            │ CVSS │ CVSS │ ISSUE ID    │
│          │ PACKAGE                                                                      │ PACKAGE │ PACKAGE    │ PACKAGE  │ VERSIONS │        │                │ V2   │ V3   │             │
│          │                                                                              │ VERSION │ NAME       │ VERSION  │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__e6f0569d0d53924e95d6c952eda94f9f7de0a22a954b2e758ef0e3d665999427.tar │         │ aiohttp    │ 3.9.1    │ [3.9.2]  │ Python │ CVE-2024-23334 │      │ 7.5  │ XRAY-589577 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__b234c65d8980a86624c28589f17b80f447e1559fffec9b40e0e4fb49a0c5502d.tar │         │ setuptools │ 65.5.1   │ [70.0.0] │ Python │ CVE-2024-6345  │      │ 8.8  │ XRAY-611082 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__b4d3e178cf4e0a6345bc893b8bb28c9b0fa56180f4bd0e13079a54e79ee48d6d.tar │         │ setuptools │ 65.5.0   │ [70.0.0] │ Python │ CVE-2024-6345  │      │ 8.8  │ XRAY-611082 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__e6f0569d0d53924e95d6c952eda94f9f7de0a22a954b2e758ef0e3d665999427.tar │         │ GitPython  │ 3.1.32   │ [3.1.33] │ Python │ CVE-2023-40590 │      │ 7.8  │ XRAY-529237 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__e6f0569d0d53924e95d6c952eda94f9f7de0a22a954b2e758ef0e3d665999427.tar │         │ GitPython  │ 3.1.32   │ [3.1.41] │ Python │ CVE-2024-22190 │      │ 7.8  │ XRAY-588270 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__e6f0569d0d53924e95d6c952eda94f9f7de0a22a954b2e758ef0e3d665999427.tar │         │ gunicorn   │ 20.1.0   │ [22.0.0] │ Python │ CVE-2024-1135  │      │ 8.2  │ XRAY-598275 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
├──────────┼──────────────────────────────────────────────────────────────────────────────┼─────────┼────────────┼──────────┼──────────┼────────┼────────────────┼──────┼──────┼─────────────┤
│ 🔥High   │ sha256__e6f0569d0d53924e95d6c952eda94f9f7de0a22a954b2e758ef0e3d665999427.tar │         │ aiohttp    │ 3.9.1    │ [3.9.4]  │ Python │ CVE-2024-30251 │      │ 7.5  │ XRAY-600100 │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
│          │                                                                              │         │            │          │          │        │                │      │      │             │
└──────────┴──────────────────────────────────────────────────────────────────────────────┴─────────┴────────────┴──────────┴──────────┴────────┴────────────────┴──────┴──────┴─────────────┘
```

___

### **PR Type**
Bug fix, Dependencies

___

### **Description**
- Updated `setuptools` version requirement in `pyproject.toml` from `>=61.0` to `>=70.0`.
- Updated `aiohttp` version from `3.9.1` to `3.9.2` to fix a high-severity security vulnerability (CVE-2024-23334).
- Updated `GitPython` version from `3.1.32` to `3.1.41`.
- Updated `gunicorn` version from `20.1.0` to `22.0.0`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update `setuptools` version requirement in `pyproject.toml`</code></dd></summary>
<hr>

pyproject.toml

<li>Updated <code>setuptools</code> version requirement from <code>>=61.0</code> to <code>>=70.0</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1083/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Update dependencies to address security vulnerabilities</code>&nbsp; &nbsp; </dd></summary>
<hr>

requirements.txt

<li>Updated <code>aiohttp</code> version from <code>3.9.1</code> to <code>3.9.2</code>.<br> <li> Updated <code>GitPython</code> version from <code>3.1.32</code> to <code>3.1.41</code>.<br> <li> Updated <code>gunicorn</code> version from <code>20.1.0</code> to <code>22.0.0</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1083/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,41 +1,41 @@
-FROM python:3.10 as base
+FROM python:3.10 AS base
```
Comment (pc-bob) [inline docker/Dockerfile:1]: fixes warnings on docker build:

```
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)                                                                                                                                                                                                                                                                                                                      0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 10)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 14)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 18)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 22)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 26)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 30)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 34)                                                                                                                                                                                                                                                                                                                     0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 40)  
```

**Comment by pc-bob** [comment]
With these changes all the vulns except the `setuptools` one should be resolved. I believe that is bundled with python and upgrading python probably has some additional implications. 

Screenshot 2024-07-31 at 3 08 35 PM

**Comment by mrT23** [comment]
thanks.
will be deployed in the nightly version

**Review by mrT23 — APPROVED** [review]

**Comment by mrT23** [comment]
deployed to `codiumai/pr-agent:github_action`

**Comment by pc-bob** [comment]
@mrT23 when can we expect a new tagged version to be released?

**Comment by mrT23** [comment]
in 2-3 weeks.
but you can use the docker SHA and get something equivalent - a **fixed** version


for example in github action:
https://pr-agent-docs.codium.ai/installation/github/#run-as-a-github-action

```
    steps:
      - name: PR Agent action step
        id: pragent
        uses: docker://codiumai/pr-agent@sha256:a4cde7a24586f88de4fbb03a0ffdb0d0e8b9f76400f6a30764159a6dcb61aad9
```

**Comment by pc-bob** [comment]
wonderful, thank you!

**Pull request merged**